pandas/core/internals/blocks.py

```diff
@@ -2499,9 +2499,9 @@ def _can_hold_element(self, element):
     def fillna(self, value, **kwargs):
 
         # allow filling with integers to be
-        # interpreted as seconds
+        # interpreted as nanoseconds
```

Comment:
this is an api breaking change, not sure its easy nor desirable to actually do this

Reply:
> not sure its easy nor desirable to actually do this

"easy" seems clear, since this PR already does it with an 11-line diff.  Why wouldn't it be desirable?  It is hugely inconsistent with how we treat ints in td64 everywhere else.

Reply:
it’s a user facing api change

Reply:
We can deprecate it pretty easily too right? Say that they should pass `.fillna(pd.Timestamp(val, unit='s'))` if they want to keep the old behavior?

Reply:
maybe we should eventually just raise on integers rather than just change the interpretation. I think thats a bit less flexibile but more type strict.

Reply:
It would allow us to use TimedeltaArray.fillna more directly

Reply:
great